src/briefcase/commands/new.py

```diff
@@ -427,28 +450,68 @@ def build_app_context(self):
             "bundle": bundle,
             "url": url,
             "license": project_license,
-            "gui_framework": (gui_framework.split())[0],
         }
 
+    def build_gui_context(self, context: dict[str, str]) -> dict[str, str]:
+        """Build context specific to the GUI toolkit."""
+        bootstraps = get_gui_bootstraps()
+        bootstrap_choices = [
+            "Toga",
+            "PySide2       (does not support iOS/Android deployment)",
+            "PySide6       (does not support iOS/Android deployment)",
+            "PursuedPyBear (does not support iOS/Android deployment)",
+            "Pygame        (does not support iOS/Android deployment)",
+        ]
+        builtin_bootstraps = [c.split(" ")[0] for c in bootstrap_choices]
+        # add choices for bootstraps that aren't built-in to Briefcase
+        bootstrap_choices.extend(set(bootstraps) - set(builtin_bootstraps))
+        bootstrap_choices.append("None")
+
+        bootstrap_choice = self.input_select(
+            intro="""
+What GUI toolkit do you want to use for this project?""",
+            variable="GUI framework",
+            options=bootstrap_choices,
+        )
+
+        gui_context = {}
+
+        if bootstrap_choice != "None":
+            try:
+                bootstrap_class = bootstraps[bootstrap_choice]
+            except KeyError:
+                bootstrap_class = bootstraps[bootstrap_choice.split(" ")[0]]
```

Comment:
"There must be a better way"

I didn't _want_ to do a bunch of refactoring here....so, I forced all this to work with what existed and behave the way it did before. There's just a bunch of cascading effects if doesn't behave like before (e.g. notably, this ordering of GUIs is necessary for all of CI work).

I think a big improvement would be string matching on the input choice; so, instead of requiring the input to be the index of the choice, allow an input of "To" to match on Toga, for instance. This is obviously non-trivial and I decided not to bite it off here.

That said, I'm open to ideas for improving.

Reply:
> If I'm understanding the problem, that _should_ address the testing issue - the list of plugins should be predictable in a CI setting. If I've missed something, let me know.

Not just predictable; the first 6 choices must be exactly known prior to CI running.

For instance, the list could be predictably alphabetical...but that isn't good enough for CI since it [injects](https://github.com/beeware/.github/blob/81b95eff757fcf485c1a8a2dfbdad41c2b97855c/.github/workflows/app-build-verify.yml#L145) the index for the option in to `stdin` to choose which GUI toolkit to use. Therefore, the list must be exactly what we have here.

This is also further complicated by the fact that "None" is last. In order for Briefcase to test the "None" case, no other plugins can be installed in to the environment....otherwise, option 6 is no longer "None". Technically, this is already a problem for running `pytest` outside of `tox`.

That's why I suggested that string matching on the name of choice instead of only the index of the option may improve this situation.

Now, admittedly...my nitpicking here over "predictable" vs "exactly known" is only relevant if other plugins are installed in to the environment since that's the only thing that could change this list from what it'd be by default. Right now, that isn't a problem for CI. However, the next step I'm planning for this work is to install a dedicated "Briefcase Automation" plugin in to CI so we can test running the helloworld apps there.

Sooo, explicitly controlling the listing of the GUI toolkits options like this is the most stable way to present this list....in that it ensures:
- The built-in plugins are always the first 5 choices in a specific order 
- Any additional plugin that's installed will be option 6
- As long as no other plugins are installed, "None" will be option 6

> If the issue is converting "3" into whichever bootstrap is third - IIRC we can pass a dictionary into input_select, and the selected option that is returned is the key from the selected choice. That's used for the Android device selector - you select item "3", but get back the device ID for the target you select. That should allow us to use the class name of the bootstrap as the key.

You're thinking of the [other](https://github.com/beeware/briefcase/blob/a0b30978380962d2eb0ac717128f9f0cf496d3da/src/briefcase/console.py#L669) "input select" method :) which takes me back actually....because one of my first PRs was an [attempt](https://github.com/beeware/briefcase/commit/e54a496406e34a1e3edf8f2ca430801e50aaebda#diff-c8b948515fc59ab4b4fe3edc9076b0c51e8c3f70ffc05e72e7e2b7a0eacdaab3) to corral input solicitation...but that was ultimately toned down.

Nonetheless, I'll see about switching this over to `briefcase.console.select_option()` which does allow the options to be a dictionary.

Reply:
Unfortunately, `briefcase.console.select_option()` isn't going to work because it doesn't respect the order of the options from the caller.

Reply:
scratch that...I remembered that the Android device list always has the "running devices" first...so I must have missed something. A closer looks shows that only choices in dictionaries are sorted; a list of tuples is not.